### ISSUE TITLE
`Credo.Check.Design.AliasUsage` doesn't warn on un-aliased struct modules

### ISSUE THREAD

**Issue by superhawk610** [rrrene/credo#1303]
Given the following code:

```elixir
defmodule Foo do
  def run do
    %Foo.Bar.Baz{..}
  end
end
```

I'd expect `Credo.Check.Design.AliasUsage` to warn on `Foo.Bar.Baz`, as it could be aliased:

```elixir
defmodule Foo do
  alias Foo.Bar.Baz

  def run do
    %Baz{..}
  end
end
```

I don't mind whether this is configurable or enabled by default, but off the top of my head I can't think of any places where I'd prefer the un-aliased form.
